pq-src/build.rs

```diff
@@ -181,90 +185,96 @@ int main() {"#,
 make_test_for!(TEST_FOR_STRCHRNUL, r#"strchrnul("", 42);"#);
 make_test_for!(TEST_FOR_STRSIGNAL, r#"strsignal(32);"#);
 
-fn check_compiles(test: &str) -> bool {
-    let test_path = std::env::var("OUT_DIR").expect("Set by cargo") + "/test.c";
-    std::fs::write(&test_path, &test).expect("Failed to write test");
-    let r = cc::Build::new().file(&test_path).try_compile("test");
-    std::fs::remove_file(test_path).expect("Failed to remove test file");
+fn check_compiles(test: &str, mut command: cc::Build) -> bool {
```

Comment:
I think we want to call [`Build::cargo_metadata(false)`](https://docs.rs/cc/latest/cc/struct.Build.html#method.cargo_metadata) for the test build (but **not** for the real libpq build, the owned `cc::Build` should already take care of that).
That should allow us to remove the `libtest.a` artifact, although that's not necessary to have.

Reply:
I did implement that (along with removing libtest.a) in e4557ce as (imo) it's cleaner that way.